tools/base/utils.py

```diff
@@ -26,3 +27,42 @@ def coverage_with_data_file(data_file: str) -> Iterator[str]:
         with open(tmprc, "w") as f:
             parser.write(f)
         yield tmprc
+
+
+class BufferUtilError(Exception):
+    pass
+
+
+@contextmanager
+def nested(*contexts):
```

Comment:
related to above comments - this *used* to be part of stdlib - beyond my grokking  why they would remove such a useful thing, but they did.

I find myself readding it to most codebases that i work on - if for no other reason, because it makes patching in tests *much* cleaner